rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -553,22 +571,14 @@ async fn run_controller(
                         }
                         controller.refresh_system_tray_menu()?;
                     }
-                    Req::CancelSignIn | Req::DisconnectedTokenExpired | Req::SignOut => {
-                        tracing::debug!("Token expired, user signed out, or user canceled sign-in");
-                        controller.auth.sign_out()?;
-                        controller.tunnel_ready = false;
-                        if let Some(mut session) = controller.session.take() {
-                            tracing::debug!("disconnecting connlib");
-                            // This is redundant if the token is expired, in that case
-                            // connlib already disconnected itself.
-                            session.connlib.disconnect(None);
-                        }
-                        else {
-                            // Might just be because we got a double sign-out or
-                            // the user canceled the sign-in or something innocent.
-                            tracing::warn!("tried to sign out but there's no session");
-                        }
-                        controller.refresh_system_tray_menu()?;
+                    Req::CancelSignIn | Req::SignOut => {
+                        tracing::info!("User signed out or canceled sign-in");
+                        controller.sign_out()?;
+                    }
+                    Req::DisconnectedTokenExpired => {
+                        tracing::info!("Token expired");
```

Comment:
Just curious -- how are you detecting that the token has expired? IIRC it's an opaque token, so in most cases a `401` means the token has expired so this is probably fine

Reply:
I match on the error type in connlib's `on_disconnect` down here https://github.com/firezone/firezone/blob/e5e10b8641e609c57f2eb6688dcc33c445f47bc9/rust/windows-client/src-tauri/src/client/gui.rs#L311-L316

Looks like it was from #3142